src/WattleScript.Interpreter/Execution/VM/Processor/Processor_InstructionLoop.cs

```diff
@@ -1463,6 +1475,14 @@ private void ExecTblInitI(Instruction i)
 			m_ValueStack.RemoveLast(i.NumVal);
 		}
 
+		private void ExecNewRange(Instruction i)
+		{
+			DynValue toDv = m_ValueStack.Pop().ToScalar();
+			DynValue fromDv = m_ValueStack.Pop().ToScalar();
+				
+			m_ValueStack.Push(DynValue.NewRange(new Range(m_Script, fromDv.CastToInt() ?? 0, toDv.CastToInt() ?? 0)));
```

Comment:
This should do a proper check with `TryCastToNumber` and throw a new exception that you can define in `ScriptRuntimeException` (see below for an example method, + see the rest of the loop to see the usual pattern for throwing exceptions). Silently assigning zero is a bad idea

```cs
/// <summary>
/// Creates a ScriptRuntimeException with a predefined error message specifying that
/// an object could not be converted
/// </summary>
/// <param name="obj">The object.</param>
/// <returns>
/// The exception to be raised.
/// </returns>
public static ScriptRuntimeException ConvertType(DataType from, DataType to)
{
return new ScriptRuntimeException("could not convert {0} to {1}", from.ToLuaTypeString(), to.ToLuaTypeString());
}
```

Reply:
Added in [ea86dbd](https://github.com/WattleScript/wattlescript/pull/36/commits/ea86dbd442ce1abc6a2f39118fec572d5ccebfb3)